src/EventListener/TagListener.php

```diff
@@ -74,12 +89,14 @@ public function onKernelResponse(FilterResponseEvent $event)
         $request = $event->getRequest();
         $response = $event->getResponse();
 
-        $tags = [];
-        // Only set cache tags or invalidate them if response is successful
-        if ($response->isSuccessful()) {
-            $tags = $this->getAnnotationTags($request);
+        if (!$this->cacheableRule->matches($request, $response)
+            && !$this->mustInvalidateRule->matches($request, $response)
```

Comment:
this would be an odd situation when neither matches. but i think there is not much else we can do.